woocommerce/woocommerce
# Fix background removal spotlight z-index display issue on Safari.

Issue:
When using the Safari browser, there's an observed behavior where the "Remove Backgrounds with AI" spotlight is rendered behind the WordPress attachments modal. This impedes the intended user guidance and impacts the overall user experience, as the spotlight remains obscured.

![Screen Capture on 2023-10-23 at 12-59-31](https://github.com/woocommerce/woocommerce/assets/6723003/57f732b3-f62e-4552-85ad-8715856c9395)

Safari manages stacking contexts differently than other browsers, which is the cause of the discrepancy seen here.

**Attempts to Resolve:**
In my journey to fix this, I tried several things:

- Adjusting the z-index of the spotlight and related components.
- Removing the spotlight element and appending it to another DOM node.
- [All kinds of hacky stuff](https://ecomgraduates.com/en-de/blogs/news/fixing-z-index-issue-on-safari-browser) to try to get Safari to render a new stacking context (nothing worked).

My proposed solution:
This PR entails modifying the the component's mounting strategy, essentially, changing its parent DOM node.
This puts it in a stacking context which resolves the issue.
I tried for the life of me to find a CSS solution but couldn't. Additionally, even when using the Layers tool within Safari, it made it appear as if this spotlight should be fully visible, but it wasn't. Perhaps we're fighting a Safari bug that isn't fixable with CSS?

![Screen Capture on 2023-10-23 at 14-52-52](https://github.com/woocommerce/woocommerce/assets/6723003/37ee968c-eca0-47f6-91c6-83903f8eb2c7)

The other part of this fix was ensuring that the Spotlight was not rendered if its anchor element doesn't exist (that would be the "Remove Background" link on the sidebar). To do this, I had to utilize a mutation observer. I would really prefer to not bloat our code with complexity but this was the best approach I could find after trying many.

However, I want to treat this PR also as an RFC as it's not without side effects. You might experience a brief "jump" or "flash" on the screen as a result of these changes.

I encourage the team to approach this solution critically. We should weigh the benefits against the side effects and consider potential alternative approaches. Feedback is sought on refining the current fix, minimizing side effects, or exploring entirely different strategies to ensure a consistent and optimal user experience across browsers.

### How to test the changes in this Pull Request:

Make sure to delete the `woo-ai-plugin-prefs` local storage key so that you can see the spotlight.

### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
